editoast/src/views/infra/mod.rs

```diff
@@ -463,14 +478,43 @@ async fn call_core_infra_load(
 
 #[post("/infra_load")]
 async fn infra_load(
-    payload: Json<Payload>,
+    payload: Json<LoadPayload>,
     db_pool: Data<DbPool>,
     core: Data<CoreClient>,
 ) -> Result<String> {
     call_core_infra_load(payload, db_pool, core).await?;
     Ok("infra loaded".into())
 }
 
+/// Builds a Core infra state request, runs it
+async fn call_core_cache_status(
+    payload: Option<Json<StatePayload>>,
+    db_pool: Data<DbPool>,
+    core: Data<CoreClient>,
+) -> Result<HashMap<String, InfraStateResponse>> {
+    let payload = payload.unwrap_or(Json(StatePayload { infra: None }));
+    if let Some(infra_id) = payload.infra {
+        Infra::retrieve(db_pool.clone(), infra_id)
+            .await?
+            .ok_or(InfraApiError::NotFound { infra_id })?;
+    };
+    let infra_request = InfraStateRequest {
+        infra: payload.infra,
+    };
+    let response = infra_request.fetch(core.as_ref()).await?;
+    Ok(response)
+}
+
+#[get("/cache_status")]
+async fn cache_status(
+    payload: Option<Json<StatePayload>>,
+    db_pool: Data<DbPool>,
+    core: Data<CoreClient>,
+) -> Result<Json<HashMap<String, InfraStateResponse>>> {
+    let response = call_core_cache_status(payload, db_pool, core).await?;
```

Comment:
Why move the logic out of the endpoint function if it leaves only two lines in the endpoint function ?

Reply:
You are right (me = no brain)